Crypto/Internal/ByteArray.hs

```diff
@@ -37,3 +40,17 @@ constAllZero b = unsafeDoIO $ withByteArray b $ \p -> loop p 0 0
             e <- peekByteOff p i
             loop p (i+1) (acc .|. e)
     len = Data.ByteArray.length b
+
+-- | Apply a function to every element of a byte array.
+-- Assumes the array is of size some integer multiple of the element.
+unsafeMapWords :: forall a ba1 ba2
+                . (ByteArrayAccess ba1, ByteArray ba2, Storable a)
+               => (a -> a) -> ba1 -> ba2
+unsafeMapWords f prevSt = unsafeDoIO $ do
+    newSt  <- copy prevSt (\_ -> return ())
+    withByteArray newSt $ \stPtr -> do
+        let sz = Data.ByteArray.length prevSt `div` sizeOf (undefined :: a)
+        for_ [0 .. pred sz] $ \i -> do
+            w <- peekElemOff stPtr i
+            pokeElemOff stPtr i (f w)
+    return newSt
```

Comment:
A single call to copyAndFreeze can replace the 3 calls to unsafeDoIO, copy, withByteArray.

Reply:
Done in 1686aee